src/operations/get-create-entity-with-file.handler.ts

```diff
@@ -0,0 +1,47 @@
+import { CreateOperationWithFileConfig } from '../typing-utils/operations';
+import { Request, Response } from 'express';
+import { RequestHandler } from 'express-serve-static-core';
+import { DataAdapterStorage } from '../utils/create-data-adapter-storage';
+import { delay, getNumberId } from '../utils/misc';
+import { genUid } from 'light-uid';
+import { createResponseBuilder } from '../utils/create-response-model';
+import { setPropDeep } from '../utils/object-utils';
+
+export const getCreateEntityWithFileHandler = (
+  methodConfigs: CreateOperationWithFileConfig,
+  dataAdapterStorage: DataAdapterStorage,
+): RequestHandler => {
+  const dataAdapter = dataAdapterStorage.getAdapter(methodConfigs.dataJsonPath!);
+  const { type, name } = methodConfigs.uidField;
+
+  return async (req: Request, res: Response) => {
+    const responseBuilder = createResponseBuilder(methodConfigs.responseModel);
+
+    const newDataItems = [] as any[];
+    (req.files as Express.Multer.File[]).forEach((file: Express.Multer.File) => {
+      let newDataItem = {
+        [name]: type === 'number' ? getNumberId(dataAdapter.getAll(), name) : genUid(),
+        ...req.body,
+        ...methodConfigs.extensions.withDefaultValues,
+      };
+
+      newDataItem = {
+        ...newDataItem,
+        ...setPropDeep(newDataItem, methodConfigs.handleFile.exportedFields?.fileName, file.originalname),
+        ...setPropDeep(newDataItem, methodConfigs.handleFile.exportedFields?.fileSize, file.size),
+      }
+
+      newDataItems.push(newDataItem);
+      dataAdapter.addOne(newDataItem, methodConfigs.save!);
+    });
+
+    await delay(methodConfigs.delay);
+
+    responseBuilder.setData(
+      methodConfigs.returnEntity
+        ? newDataItems.length === 1 ? newDataItems[0] : newDataItems
+        : newDataItems.length === 1 ? { [name]: newDataItems[0][name] } : newDataItems.map((dataItem) => ({ [name]: dataItem[name] }))
+    );
```

Comment:
do the mapping and then picking based on returnEntity. That'll make your life easier.

Reply:
Also, sometimes it should be multiple anyway. I think you need some other option telling you if that has to be one or multiple.